pkg/cnab/provider/driver.go

```diff
@@ -44,6 +48,40 @@ func (r *Runtime) newDriver(driverName string, args ActionArguments) (driver.Dri
 		return nil, err
 	}
 
+	// Handle Directory support for the docker driver
+	// TODO: Handle directory support for other runtimes -- how?
```

Comment:
Here's what I suggest:

1. Do not encode mount information into the directory parameter extension other than the information from `DirectoryParameterDefinition`. The rest is docker specific and should be specified by the user running the bundle, not the bundle author.
2. Check if docker is the driver, then handle the directory parameter just like you are now but... change where you get the information about the mount. Iterate over the bundle's parameters and set up mounts for each parameter of type directory.
3. Use the parameter's value as a docker --mount flag formatted string to get the mount details. The target will come from the parameter definition itself.

The extension data will need to provide details about each directory parameter. (will comment more about that on the data structures in question)